### PR TITLE
Deprecate {ion,neutral_atom}.decompose_operation.

### PR DESCRIPTION
**Pull request by MichaelBroughton**
Yet more of #4744

### CODE REVIEW

**Review by vtomole — APPROVED** [review]

File: cirq-core/cirq/ion/ion_device.py [hunk 1]
```diff
@@ -87,6 +87,9 @@ def qid_pairs(self) -> FrozenSet['cirq.SymmetricalQidPair']:
             qs = self.qubits
             return frozenset([devices.SymmetricalQidPair(q, q2) for q in qs for q2 in qs if q < q2])
 
+    @_compat.deprecated(
+        fix='Use ConvertToIonGates() instead to decompose operations.', deadline='v0.15'
```
Comment (tanujkhattar) [inline cirq-core/cirq/ion/ion_device.py:91]: s/`ConvertToIonGates()`/`cirq.ConvertToIonGates()`

File: cirq-core/cirq/neutral_atoms/neutral_atom_devices.py [hunk 2]
```diff
@@ -122,6 +122,9 @@ def qubit_set(self) -> FrozenSet['cirq.GridQubit']:
     def qubit_list(self):
         return [qubit for qubit in self.qubits]
 
+    @_compat.deprecated(
+        fix='Use ConvertToNeutralAtomGates() instead to decompose operations.', deadline='v0.15'
```
Comment (tanujkhattar) [inline cirq-core/cirq/neutral_atoms/neutral_atom_devices.py:126]: s/`ConvertToNeutralAtomGates()`/`cirq.ConvertToNeutralAtomGates()`

**Review by tanujkhattar — APPROVED** [review]
LGTM % nits

File: cirq-core/cirq/ion/ion_device.py (hunk 1 above)
Comment (MichaelBroughton) [reply cirq-core/cirq/ion/ion_device.py]: Done.

File: cirq-core/cirq/neutral_atoms/neutral_atom_devices.py (hunk 2 above)
Comment (MichaelBroughton) [reply cirq-core/cirq/neutral_atoms/neutral_atom_devices.py]: Done.

**Pull request merged**